### ISSUE TITLE
Page Builder Products Element not works

### ISSUE THREAD

**Issue by brmor-mistermenuiserie** [Smile-SA/elasticsuite#2510]
### Preconditions
Magento 2.4.1-p1
ElasticSuite Version : 2.10.9.1
Environment : Production

### Steps to reproduce

1. create a Cms page with Page builder Content and Products Element
2. select Product By Sku
3. Enter Skus into Products SKUs field
4. Choose Sort By Position

### Expected result

1. We need to see selected product on Frontend

### Actual result

1. Any Products are visible on frontend.

If we choose position other than Position (position_by_sku) it works.
Seems to be due to your Plugin ProductsListPlugin function afterCreateCollection.

Also we have this issue only on production Mode, in our local environment (dev mode) it works

**Comment by romainruaud** [comment]
Hi, we will try to reproduce this one.

Are you able to give me an output of the Elasticsearch query that is generated for this widget ? You should be able to see it in the logs after enabling debug mode.

Thanks

**Comment by rensieeee** [comment]
Hi,

We had a similar issue. We noticed this was a result of the html contents of the category page being too big to index. If you remove parts of the page builder content and reindex, the products should be visible correctly.

**Comment by brmor-mistermenuiserie** [comment]
> ```
> [2022-04-07 14:07:23] report.WARNING: Response ["{\"error\":{\"root_cause\":[{\"type\":\"illegal_argument_exception\",\"reason\":\"cannot execute [inline] scripts\"}],\"type\"
> ```
> 
> I guess it's a Magento Cloud hosting, you'll have to ask them to add these lines to the configuration :
> 
> ```
> script.allowed_types: inline
> script.allowed_contexts: search,update,aggs
> ```

Ok, So I will ask it to Magento Cloud.
Maybe, I'm wrong, but it's not describe on your documentation ?

**Comment by romainruaud** [comment]
You are right, this was not documented. Script sorting is only used for sorting products by position on the product widget, so it's a recent addition.

But this is a configuration that is enabled by default on any Elasticsearch server, I don't know why it's not the case on Magento Cloud.

Anyway, I added it to the documentation : https://github.com/Smile-SA/elasticsuite/wiki/MeceInstall#ensure-proper-elasticsearch-configuration-is-set-for-inline-scripts
